src/checkers/inference/dataflow/InferenceValue.java

```diff
@@ -64,14 +64,16 @@ public CFValue leastUpperBound(CFValue other) {
         Slot slot2 = getEffectiveSlot(other);
 
         if (slot1 instanceof ConstantSlot && slot2 instanceof ConstantSlot) {
+            AnnotationMirror anno1 = slotManager.getAnnotation(slot1);
+            AnnotationMirror anno2 = slotManager.getAnnotation(slot2);
             final AnnotationMirror
-                    lub = qualifierHierarchy.leastUpperBound(((ConstantSlot) slot1).getValue(), ((ConstantSlot) slot2).getValue());
+                    lub = qualifierHierarchy.leastUpperBound(anno1, anno2);
```

Comment:
Add comment to clarify what qualHier is here, what annoType are of anno1 and anno2.